### ISSUE TITLE
Cannot compile with 7.10

### ISSUE THREAD

**Issue by crockeea** [konn/monomorphic#6]
I'm getting the following error:

```
Data/Type/Monomorphic.hs:45:17:
    Couldn't match type ‘k2 a0 -> b’ with ‘forall (a :: k0). k1 a -> b’
    Expected type: MonomorphicRep k2 -> (k2 a0 -> b) -> b
      Actual type: MonomorphicRep k1
                   -> (forall (a :: k0). k1 a -> b) -> b
    Relevant bindings include
      liftPoly :: (forall (a :: k). k2 a -> b) -> MonomorphicRep k2 -> b
        (bound at Data/Type/Monomorphic.hs:45:1)
    In the first argument of ‘flip’, namely ‘withPolymorhic’
    In the expression: flip withPolymorhic
```

I changed line 45 from `liftPoly = flip withPolymorphic` to  `liftPoly trans k = withPolymorhic k trans`, and everything compiles. This seems like suspicious behavior, possibly a 7.10 bug?

**Comment by crockeea** [comment]
Asked and answered [here](http://stackoverflow.com/questions/29712710/polymorphic-flip-fails-in-7-10/).

**Comment by Athas** [comment]
I have a pull request that solves this.

**Comment by konn** [comment]
I forgot to close this issue. This was fixed via d8db2fa908.
